internal/stage_magnet_parse.go

```diff
@@ -0,0 +1,58 @@
+package internal
+
+import (
+	"fmt"
+
+	"github.com/codecrafters-io/tester-utils/test_case_harness"
+)
+
+func testParseMagnetLink(stageHarness *test_case_harness.TestCaseHarness) error {
+	type MagnetLink struct {
+		UrlEncoded string
+		Trackers []string
+		InfoHash string
+	}
+
+	magnetLinks := []MagnetLink {
+		{
+		   UrlEncoded: "magnet:?xt=urn:btih:c77829d2a77d6516F88cd7a3de1a26abcbfab0db&dn=codercat.gif&tr=http%3A%2F%2Fbittorrent-test-tracker.codecrafters.io%2Fannounce",
+		   Trackers: []string { "http://bittorrent-test-tracker.codecrafters.io/announce" },
+		   InfoHash: "c77829d2a77d6516F88cd7a3de1a26abcbfab0db",
+	   },
+	   {
+		UrlEncoded: "magnet:?xt=urn:btih:c77829d2a77d6516F88cd7a3de1a26abcbfab0db",
```

Comment:
Do we necessarily have to test this case? Doesn't look like we're going to test any such "tracker-less" links in later stages. Can we just make all tests here have at least the info hash and tracker URL?

Reply:
Yep, if our goal is to ensure that people can pass all stages, we don't need to test this case
My goal was to cover edge cases people might see in the real world, like no trackers, multiple trackers, malformed link, extra parameters etc. because otherwise we get the criticism that it feels like a toy project and not the real thing ...